ably/rest_client.go

```diff
@@ -35,6 +36,23 @@ func query(fn func(context.Context, string, interface{}) (*http.Response, error)
 	}
 }
 
+type Raw []byte
+
+func (r *Raw) UnmarshalJSON(data []byte) error {
```

Comment:
This function signature here can return an error, but I can see the code always returns nil. Is there some error handling missing here? Unmarshalling JSON is usually something that can go wrong and generate an error. How would that error be returned?

Reply:
The raw type just pretends to decode but returns the original bytes.

https://pkg.go.dev/encoding/json#RawMessage

Will be commented as  I forgot to comment Raw.

Reply:
I see, so `Raw` has this method so that it can pass through the `Unmarshaler` interface.